homeassistant/helpers/helper_config_entry_flow.py

```diff
@@ -79,6 +79,23 @@ async def _async_form_step(
         """Handle a form step."""
         form_step: HelperFlowFormStep = cast(HelperFlowFormStep, self._flow[step_id])
 
+        if (
+            user_input is not None
+            and (data_schema := form_step.schema)
+            and data_schema.schema
+            and not self._handler.show_advanced_options
+        ):
+            # Add advanced field default if not set
+            for key in data_schema.schema.items():
+                if isinstance(key, (vol.Optional, vol.Required)):
+                    if (
+                        key.description
+                        and key.description.get("advanced")
+                        and key.default is not vol.Undefined
+                        and key not in self._options
+                    ):
+                        user_input[key] = key.default
+
```

Comment:
This is not needed for the group flow once https://github.com/home-assistant/core/pull/68609 is merged because `all` can then not be set in the config flow.